config/vlan.py

```diff
@@ -189,97 +231,159 @@ def vlan_member():
+
+            # default vlan checker
+            if vid == 1:
+                ctx.fail("{} is default VLAN".format(vlan))
+
+            log.log_info("'vlan member add {} {}' executing...".format(vid, port))
+
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist".format(vlan))
+
+            if clicommon.get_interface_naming_mode() == "alias":  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                alias = port
+                iface_alias_converter = clicommon.InterfaceAliasConverter(db)
+                port = iface_alias_converter.alias_to_name(alias)
+                if port is None:
+                    ctx.fail("cannot find port name for alias {}".format(alias))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.is_port_mirror_dst_port(db.cfgdb, port):
+                ctx.fail("{} is configured as mirror destination port".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.is_port_vlan_member(db.cfgdb, port, vlan):
+                log.log_info("{} is already a member of {}, Aborting!!!".format(port, vlan))
+                ctx.fail("{} is already a member of {}, Aborting!!!".format(port, vlan))
+                
+
+            if clicommon.is_valid_port(db.cfgdb, port):
+                is_port = True
+            elif clicommon.is_valid_portchannel(db.cfgdb, port):
+                is_port = False
+            else:
+                ctx.fail("{} does not exist".format(port))
+            
+            if (is_port and clicommon.is_port_router_interface(db.cfgdb, port)) or \
+                (not is_port and clicommon.is_pc_router_interface(db.cfgdb, port)): # TODO: MISSING CONSTRAINT IN YANG MODEL
+                ctx.fail("{} is in routed mode!\nUse switchport mode command to change port mode".format(port))
+
+            portchannel_member_table = db.cfgdb.get_table('PORTCHANNEL_MEMBER')
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if (is_port and clicommon.interface_is_in_portchannel(portchannel_member_table, port)):
+                ctx.fail("{} is part of portchannel!".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if (clicommon.interface_is_untagged_member(db.cfgdb, port) and untagged):
+                ctx.fail("{} is already untagged member!".format(port))
+            
+            # checking mode status of port if its access, trunk or routed
+            if is_port:
+                port_data = config_db.get_entry('PORT',port)
+            
+            # if not port then is a port channel
+            elif not is_port:
+                port_data = config_db.get_entry('PORTCHANNEL',port)
+
+            if "mode" not in port_data: 
```

Comment:
Hi @sabakram , I suspect this line break backward compatibility.
It will wrongly treat a port as routed when 'mode' is not present even though it pass the check in 
Line 297: 
```
if (is_port and clicommon.is_port_router_interface(db.cfgdb, port)) or \
                (not is_port and clicommon.is_pc_router_interface(db.cfgdb, port)): # TODO: MISSING CONSTRAINT IN YANG MODEL
                ctx.fail("{} is in routed mode!\nUse switchport mode command to change port mode".format(port))
```

Reply:
Hi @wen587 ,
We are already looking into backward compatibility issues for that we have removed all changes in db_migrator where we remove this default mode routed behavior.